examples/perplexity/perplexity.cpp

```diff
@@ -174,12 +240,15 @@ void perplexity_v2(llama_context * ctx, const gpt_params & params) {
         fflush(stdout);
     }
     printf("\n");
+
+    return std::make_tuple(tokens, logit_history, prob_history, std::exp(nll / count));
 }
 
-void perplexity(llama_context * ctx, const gpt_params & params) {
+std::tuple<std::vector<llama_token>, std::vector<float>, std::vector<float>, float>
```

Comment:
I find `std::tuple` syntax to be quite cumbersome and unreadable, so I normally avoid it by using a simple `struct`.

Can we change these to `struct results_ppl` and earlier to `struct `results_log_softmax`?